opencog/spacetime/Temporal.cc

```diff
@@ -64,8 +64,8 @@ void Temporal::init(unsigned long a, unsigned long b, bool normal) throw (Invali
             throw InvalidParamException(TRACE_INFO,
                                         "Cannot create a Temporal (normal-distribution) with the variance (%lu) greater than the mean (%lu). This causes negative lower bound.", b,  a);
         }
-        unsigned long long sum = (unsigned long long) a + b;
-        if (sum > (unsigned long long) UINT_MAX) {
+        octime_t long sum = (octime_t long) a + b;
```

Comment:
Hmm. It does compile, but you are right that that isn't right. In order to avoid using an unsigned long long, we could do this to prevent overflow:

octime_t sum = a/2 + b/2;
if (sum > UINT_MAX/2) { ... }

How does that sound?